sdk/core/core-http/src/util/xml.ts

```diff
@@ -2,23 +2,23 @@
 // Licensed under the MIT license.
 
 import * as xml2js from "xml2js";
+import { XML_ATTRKEY, XML_CHARKEY, SerializerOptions } from "./serializer.common";
 
 // Note: The reason we re-define all of the xml2js default settings (version 2.0) here is because the default settings object exposed
 // by the xm2js library is mutable. See https://github.com/Leonidas-from-XIV/node-xml2js/issues/536
 // By creating a new copy of the settings each time we instantiate the parser,
 // we are safeguarding against the possibility of the default settings being mutated elsewhere unintentionally.
-const xml2jsDefaultOptionsV2 = {
+const xml2jsDefaultOptionsV2: xml2js.OptionsV2 = {
   explicitCharkey: false,
   trim: false,
   normalize: false,
   normalizeTags: false,
-  attrkey: "$",
-  charkey: "_",
+  attrkey: XML_ATTRKEY,
   explicitArray: true,
   ignoreAttrs: false,
   mergeAttrs: false,
   explicitRoot: true,
-  validator: null,
+  validator: undefined,
```

Comment:
if we're setting these to undefined, should we instead omit the keys entirely? Or would that do something different?

Reply:
This is unrelated to the fix, but I added the type `xml2js.OptionsV2` and `undefined` are correct according to the type definitions. I believe this is a copy of all the supported xml2js settings so I will keep them listed here even we don't use any of them directly.